pkg/k8sutil/selector.go

```diff
@@ -5,9 +5,20 @@ import (
 
 	"k8s.io/apimachinery/pkg/fields"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
 	v1api "k8s.io/client-go/pkg/api/v1"
 )
 
+// NewRequirementOrDie wraps a call to NewRequirement and panics if the Requirment
+// cannot be created. It is intended for use in variable initializations only.
+func NewRequirementOrDie(key string, op selection.Operator, vals []string) *labels.Requirement {
```

Comment:
There is a method in the field kubernetes package called `ParseSelectorOrDie` that I was emulating with this call, since we use that one to construct selectors as well. 

As for the explicit arguments vs the return, I felt like it was cleaner to just have one function call vs two, and the type system will trivially catch any issues with modifications to the underlying constructor that we would have to care about in this function.